File: python/165. Compare Version Numbers.py
class Solution(object):
    def compareVersion(self, version1, version2):
        """
        :type version1: str
        :type version2: str
        :rtype: int
        """
        v1 = version1.split('.')
        v2 = version2.split('.')
        l1, l2 = len(v1), len(v2)
        i1 = i2 = 0
        while i1 < l1 and i2 < l2:
            a = int(v1[i1])
            b = int(v2[i2])
            if a != b:
                return a > b and 1 or -1
            i1 += 1
            i2 += 1
        
        while i1 < l1:
            if int(v1[i1]) != 0:
                return 1
            i1 += 1
            
        while i2 < l2:
            if int(v2[i2]) != 0:
                return -1
            i2 += 1
        return 0

File: go/src/leetcode/165. Compare Version Numbers/165. Compare Version Numbers.go
package main

import (
	"strconv"
	"strings"
)

func compareVersion(version1 string, version2 string) int {
	s1, s2 := strings.Split(version1, "."), strings.Split(version2, ".")
	for i := 0; i < max(len(s1), len(s2)); i++ {
		var t1, t2 int
		if i < len(s1) {
			t1, _ = strconv.Atoi(s1[i])
		}
		if i < len(s2) {
			t2, _ = strconv.Atoi(s2[i])
		}
		if t1 < t2 {
			return -1
		} else if t1 > t2 {
			return 1
		}
	}
	return 0
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}
